### PR TITLE
Add dep management directory. Ref it from PS w/o Node.

### PR DESCRIPTION
**Pull request by chexxor**
Motivated by https://github.com/purescript/documentation/issues/109

Rather than keeping guides up-to-date, I'd prefer to DRY the docs by directing the reader to follow the "how to do dep management with X" guide.

Also, I think we need an overview of the dependency management options available to PS users.

Bundling was mentioned in that "PS without Node" article. There are many options for doing that, so I'd like to give an overview of the options and give an introduction to each tool in a page for that specific tool.

LMK what you think about this direction.

### CODE REVIEW

**Comment by MiracleBlue** [comment]
I love it!  Glad that someone has taken the time to do this :) thank you!

File: guides/PureScript-Without-Node.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
-The PureScript community has standardized on the NodeJS toolset to a certain extent: we use tools like Pulp, Browserify and Gulp, which run on NodeJS and are installed with NPM, and we use Bower for our package management. However, we have some users who would rather not, or cannot, run NodeJS on their machines. In this short post, I'll demonstrate how to use the PureScript compiler directly on the command line, and how to automate some processes using standard tooling.
+
+The PureScript community has standardized on tools like Pulp, Bower, Browserify, Webpack, and RollupJS, all of which use the NodeJS runtime and use NPM as their supported distribution channel. There are users who cannot, or prefer not to, use NodeJS-based tools. PureScript can be used without NodeJS-based tools.
```
Comment (garyb) [inline guides/PureScript-Without-Node.md]: I don't think we're ready to mention rollup here... I am yet to have it work successfully on a PS project.

Comment (chexxor) [reply guides/PureScript-Without-Node.md]: I've been trying rollup for some PS projects with success. The rollup-plugin-purs thing has bitrotted a bit and needs updating, so the great tree-shaking isn't ready to talk about yet, but vanilla rollup with the rollup-plugin-commonjs and rollup-plugin-node-resolve plugins work fine.

File: guides/PureScript-Without-Node.md [hunk 2]
```diff
@@ -22,58 +23,30 @@ main = log "Hello, World!"
...
+It is necessary to provide all source files explicitly to the PureScript compiler. The simplest way is to use a set of file path globs.
 
 ```text
-$ psc-bundle output/*/{index,foreign}.js --module Main --main Main
+$ purs compile 'src/**/*.purs' 'ps-deps/purescript-*/src/**/*.purs'
 ```
 
-The `--module` argument specifies an _entry-point_ module, which will be used to determine dead code which can be removed. For our purposes, we only care about bundling the CommonJS module corresponding to the Main PureScript module and its transitive dependencies.
-
-The `--main` argument will make the output JavaScript file an executable by adding a line of JavaScript to it which runs the `main` function in the specified module.
```
[the remark this answers is not in the document]
Comment (chexxor) [reply guides/PureScript-Without-Node.md]: Yeah, I was planning to move that to an "bundling with psc-bundle" introduction doc page, then link to that from here.

File: guides/Bundling-for-Web-Browser.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+## Module Formats
+
+Bundlers can export the bundle in one of several JavaScript module formats. It's also important to know that some bundlers only support importing an application using a specific module format, though you can apply a plugin to add support for other module formats.
+
+Currently, the PureScript compiler outputs CommonJS-formatted JavaScript modules. Other JavaScript module formats include AMD, UMD, ES6, IIFE, and SystemJS. Following is a quick overview of the various module formats.
+
+The CommonJS format offers synchronous module loading and is the de facto module format of modules used by the NodeJS ecosystem.
+
+The AMD format was invented to support loading modules in a web browser asynchronously across the network.=
```
Comment (anttih) [inline guides/Bundling-for-Web-Browser.md]: Extra `=` here at the end.

File: guides/Bundling-for-Web-Browser.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+See the documentation provided by those tools for full information of how to use them. As an introduction, we'll see a basic example of these tools.
+
+Note that these tools will simply bundle a JavaScript file's dependencies into that file. A JavaScript file produced by a PureScript module will only define functions, not execute them. If you pass such a file into a bundler and simply load it into a browser, you might be disappointed to see that your project's `main` function isn't executed. To execute it in the browser requires a JavaScript file which imports the `main` function and executes it. The make your project executable, follow the bundler examples below while replacing `output/Main/index.js` with a JavaScript which executes your project. Following is an example JavaScript file which does this, here called `main-runner.js`.
```
Comment (anttih) [inline guides/Bundling-for-Web-Browser.md]: Typo "The make your project...".

File: guides/Dependency-Management/Git.purs [hunk 5]
```diff
@@ -0,0 +1,31 @@
+# Using Git for PureScript Dependency Management
```
Comment (anttih) [inline guides/Dependency-Management/Git.purs]: Wrong file extension for this file.

**Comment by chexxor** [comment]
I don't think this repo is a good place for these docs, so I'd like to close this and find a better place for it. This documentation project isn't about how to fit PureScript into your project or how to use tools to manage PureScript-using projects.

**Pull request closed without merging**